### ISSUE TITLE
최근 개선한 훈련/검증 데이터의 정확도와 손실률 예측 값 그래프

### ISSUE THREAD

**Issue by RyanKor** [RyanKor/dev-match-ML#10]
그 동안 CNN도 제대로 이해하지 못하고 적용했다는 생각이 많이 든다.

![image](https://user-images.githubusercontent.com/40455392/136897051-f36fabe1-304d-4e6d-9042-26841ee84cb0.png)

![image](https://user-images.githubusercontent.com/40455392/136897083-356666fc-5729-417f-93ce-ffa603f86b0e.png)

프로젝트 시행 착오를 겪으면서, 오버 피팅이 발생하고 있는 것을 꾸준히 해결해주면서 (합성곱 신경망 층 줄이기, Dropout 적용, BatchNormalization 사용) 위와 같이 개선을 해 볼 수 있었다.

그런데 막연히 이상한 부분은 그래프가 에폭을 반복하면서 조금도 흔들리지 않는 것이 가능한가? 에 대한 점

아무리 좋은 모델도 조금씩은 흔들렸던 것 같은데, 이 학습 정확도가 맞나 싶다.

그리고 학습 정확도와는 별개로 label이 되어 있지 않는 test dataset에 대해 정확도가 무척 낮다.

![image](https://user-images.githubusercontent.com/40455392/136897325-3a9052dc-9984-4b73-a760-5ededdd2191a.png)

검색어를 뭐로 찾아야할지 모르겠는데, 여기서 정확도가 무척 떨어져서 접근을 어떻게 해줘야할지 감이 안온다.

생활코딩이나 TF-KR, Keras Korea에 문의 해봤으나 답장은 없고,

대략적인 예측되는 에러는

1. 지나치게 훈련 / 검증 데이터에 과적합되어 있어 새로운 데이터에 대해 정확한 예측을 못함
2. 모델링이 잘못 되었다?
    ```
    model.add(Conv2D(64,(3,3),activation='relu'))
    model.add(MaxPool2D((2,2)))
    ```
    - 이 ConvNet을 2번 사용하는 것만으로는 정확한 결과 값을 예측할 정도로 모델의 성능이 나오지 않는다?
    - 위 모델은 MNIST CNN 분류 모델 예제에 소개된 값을 그대로 적용한 것인데, 이걸로 정확한 예측이 되는 것인지 일단 의문이 든다.\
3. 전이 학습 등 다른 모델을 사용할 필요성이 있다?

개인적으론 CNN에서 프로그래머스에 결과 제출 했을 때 50점 정도는 나와야 다른 모델을 적용해 볼 수 있을 것 같은데 (어느 정도 예측 정확도가 나오고 있다는 것을 확인했으므로) 테스트 예측 정확도가 어떻게 해야 개선이 될 지 충분히 고민해보자.

어떻게 찾아야할지 검색어에 대한 힌트가 필요한데, 감이 안잡힌다.

**Comment by RyanKor** [comment]
만약에 지금 작성한 코드가 새롭게 유입된 데이터의 feature를 정확하게 찾지 못해 이러한 문제가 발생한 것이라면?

근데 그럼 오히려 이해가 안되는 부분이 이전에 ResNet50 API를 따와서 제작한 모델은 왜 테스트 데이터에 대한 정확한 분류를 하지 못했느냐이다.

나와 비슷한 생각을 한 사람이 Xception? 모델을 사용해 동일한 과제를 텐서플로우로 수행한 블로그 글이 있는데 78점을 받았다고 한다.

그 사람의 model.fit을 보면 정확도도 나랑 엇비슷한데 무엇이 문제일까?

---

그리고 모델 파라미터가 너무 많으면 오히려 훈련/검증 정확도가 떨어진다.

이 프로젝트에서는 모델이 가벼워야 분류가 좀 더 잘되는 것 같다는 생각이 들어서 파라미터를 늘리는 방향보다 feature를 잘 잡아내는 방향으로 접근하자.
